File: tests/conformance/runners/java/src/test/java/fyi/sererr/conformance/ConformanceWorld.java
package fyi.sererr.conformance;

import fyi.sererr.CapturedError;
import fyi.sererr.DebugInfo;

import java.util.List;

/**
 * Per-scenario state shared across step definitions.
 *
 * <p>Picked up by Cucumber-JVM's PicoContainer DI: any step class
 * with this as a constructor parameter receives a fresh instance per
 * scenario, mirroring the Rust runner's `World` struct.
 */
public class ConformanceWorld {
    /** Constructed captured-error chain under test. */
    public List<CapturedError> chain = List.of();
    /** Captured DebugInfo for the adapter scenarios. */
    public DebugInfo debugInfo;
    /** Encoded proto bytes (when serialization is exercised). */
    public byte[] encoded;
    /** Active fixture name. */
    public String fixture;
    /** Constructed CapturedError from a fixture (pre-encode). */
    public CapturedError fixtureInput;
}

File: tests/conformance/runners/kotlin/src/test/kotlin/fyi/sererr/conformance/ConformanceWorld.kt
package fyi.sererr.conformance

import fyi.sererr.kotlin.CapturedError
import fyi.sererr.kotlin.DebugInfo

/**
 * Per-scenario state shared across step definitions.
 *
 * Cucumber-JVM's PicoContainer DI injects a fresh instance per
 * scenario when a step class accepts this in its constructor —
 * mirroring the Java runner's [fyi.sererr.conformance.ConformanceWorld].
 */
class ConformanceWorld {
    var chain: List<CapturedError> = emptyList()
    var debugInfo: DebugInfo? = null
    var encoded: ByteArray? = null
    var fixture: String? = null
    var fixtureInput: CapturedError? = null
}
